### PR TITLE
Document for assume canonical hive partition key

### PR DESCRIPTION
**Pull request by allendang001**

### CODE REVIEW

**Comment by findepi** [comment]
Does this supersede https://github.com/trinodb/trino/pull/6438 ?

File: docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -370,6 +370,23 @@ Property Name                                                   Description
...
+                                                                be present, or it may fail entirely. Thus, you should only
+                                                                set this value to true if you are certain that your
+                                                                partition keys are always stored in canonical form.
+                                                                As background, the Hive metastore uses strings for
+                                                                partition keys for all data types, but does not enforce a
+                                                                canonical representation of values. For example, a
+                                                                partition column of type int can have the value 42, but be
+                                                                stored as '0042' in the metastore. If a query has a filter
+                                                                on value 42, then the Hive connector will ask the metastore
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:386]: connectors asks the metastore

File: docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -370,6 +370,23 @@ Property Name                                                   Description
...
+                                                                for partitions matching the string '42', which will fail to
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:387]: which fails to ..

**Review by mosabua — CHANGES_REQUESTED** [review]
A bit lengthy but ok after the change to not use future tense

**Comment by bitsondatadev** [comment]
:wave: @allendang001 - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by bitsondatadev** [comment]
Gonna close this one out due to inactivity, please reopen if you would like to continue doing work here.

**Pull request closed without merging**